### PR TITLE
Fix Notebook Connection Issues

### PR DESCRIPTION
**Pull request by chlafreniere**
We weren't setting a uri prefix for every type of notebook connection, which resulted in the connectionStatusManager incorrectly trying to compare the two URIs on connection complete:

Notebook-0
file:Notebook-0

This resulted in connections not completing correctly, and notebook connections being unusable in many cases.

Ensuring that we always have the notebook uri prefix results in correct comparisons.

(sorry if I'm being vague, ping me for questions 😄)

### CODE REVIEW

File: src/sql/workbench/services/notebook/common/sql/sqlSessionManager.ts [hunk 1]
```diff
@@ -174,6 +176,7 @@ class SqlKernel extends Disposable implements nb.IKernel {
 	) {
 		super();
 		this.initMagics();
+		this.setConnectionPath();
```
Comment (alanrenmsft) [inline src/sql/workbench/services/notebook/common/sql/sqlSessionManager.ts]: is it possible to go to all the places that consumes this and make sure the path passed in is a valid URI.
the risk of doing this is you are modifying the passed value, but the caller might rely on the value for something else.

Comment (chlafreniere) [reply src/sql/workbench/services/notebook/common/sql/sqlSessionManager.ts]: Thanks. I haven't seen any ill effects here,  and validated that we only set/read this value in the SqlKernel, as the connection that we use in conjunction with the queryRunner is only referenced here. Other connections are not impacted.

Tested the following:
- Queries succeed both for SQL and Spark kernels
- IntelliSense works for one + multiple cells. Switching between cells results in correct IntelliSense
- Connections get cleaned up when notebook is closed (both that query runner uses + IntelliSense)

File: src/sql/workbench/services/notebook/common/sql/sqlSessionManager.ts [hunk 2]
```diff
@@ -183,6 +186,29 @@ class SqlKernel extends Disposable implements nb.IKernel {
...
+					let uri = URI.parse(this._path);
+					if (uri && uri.scheme) {
+						this._connectionPath = uri.toString().replace(uri.scheme, uriPrefixes.notebook);
+					}
+				} catch {
+					// Ignore exceptions from URI parsing
+				} finally {
+					// If _connectionPath hasn't been set yet, set _connectionPath to _path as a last resort
+					if (!this._connectionPath) {
+						this._connectionPath = this._path;
```
Comment (Charles-Gagnon) [inline src/sql/workbench/services/notebook/common/sql/sqlSessionManager.ts]: You could just set this value initially and then it'll just be overwritten if that code runs

File: src/sql/workbench/services/notebook/common/sql/sqlSessionManager.ts [hunk 3]
```diff
@@ -183,6 +186,29 @@ class SqlKernel extends Disposable implements nb.IKernel {
 		}
 	}
 
+	private setConnectionPath(): void {
+		if (this._path) {
+			let prefix = getUriPrefix(this._path);
+			if (!prefix || prefix === uriPrefixes.connection) {
+				this._connectionPath = uriPrefixes.notebook.concat(this._path);
```
Comment (Charles-Gagnon) [inline src/sql/workbench/services/notebook/common/sql/sqlSessionManager.ts]: It's fine to have two prefixes here?

Comment (chlafreniere) [reply src/sql/workbench/services/notebook/common/sql/sqlSessionManager.ts]: If it doesn't have a prefix, getUriPrefix returns uriPrefixes.default:

`               let index = ownerUri.indexOf('://');

		if (index > 0) {

			prefix = ownerUri.substring(0, index + 3);

		} else {

			return uriPrefixes.default;
		}
`

**Pull request merged**